### ISSUE TITLE
Block operator and dict syntax do not coexist.

### ISSUE THREAD

**Issue by jburse** [SWI-Prolog/swipl-devel#646]
I get this result:

```
?- op(200,yf,{}).
true.

?- X = f {a:b}, write_canonical(X), nl.
{}({:(a,b)},f)
X = f{a:b}.

?- X = f{a:b}, write_canonical(X), nl.
f{a:b}
X = f{a:b}.
```
The parsing is correct and gives different terms, but the unparsing not really. 
Both terms are unparsed as the same output text `X = f{a:b}` and `X = f{a:b}`.

**Comment by jburse** [comment]
I have a similar problem in my system with the new higher
order syntax. But the unparser injects a space, so the output

text is not the same:

?- op(200,yf,()).
Yes

?- X = f (a,b), write_canonical(X), nl.
()(f, a, b)
X = f (a, b)

?- X = f(a,b), write_canonical(X), nl.
f(a, b)
X = f(a, b)

**Comment by JanWielemaker** [comment]
Thanks.  Should be fixed with 6be81661f558f362253c24c77f11f6e8405685c2
